signer/core/signed_data.go

```diff
@@ -476,10 +482,54 @@ func (typedData *TypedData) EncodeData(primaryType string, data map[string]inter
 	return buffer.Bytes(), nil
 }
 
+func parseIntegerType(encType string, encValue interface{}) (*big.Int, error) {
+	length := 0
+	if encType == "int" || encType == "uint" {
+		length = 256
+	} else {
+		lengthStr := ""
+		if strings.HasPrefix(encType, "uint") {
+			lengthStr = strings.TrimPrefix(encType, "uint")
```

Comment:
btw if I try `parseIntegerType("uint32", "-123")` I get `-123`, is that what you expect?

Reply:
Nope, nice catch!